src/bp/core/services/middleware/event-engine.ts

```diff
@@ -195,6 +196,10 @@ export class EventEngine {
     }
   }
 
+  private isIncomming(event: sdk.IO.Event): event is sdk.IO.IncomingEvent {
```

Comment:
This is used only once and doesn't bring any value. I suggest keeping `event.direction === 'incoming' ` instead... (also, theres a typo in incoming)

Reply:
:+1: